server/routes/incidentDetails/incidentDetailsPage.ts

```
@@ -0,0 +1,728 @@
+            postValues.originalPageReferrerUrl ? `?referrer=${postValues.originalPageReferrerUrl}` : ''
+          }`
+        } else {
+          returnUrl = `/incident-details/${postValues.prisonerNumber}/${postValues.draftId}/edit`
+        }
+      } else {
+        returnUrl = `/incident-details/${postValues.prisonerNumber}`
+      }
+      stashDataOnSession(returnUrl, dataToSaveAfterRedirect, req)
+      if (postData === PageRequestAction.DELETE_PRISONER) {
+        return redirectToDeletePersonPage(res, postValues.incidentDetails.currentAssociatedPrisonerNumber)
+      }
+      if (postData === PageRequestAction.SEARCH_FOR_PRISONER) {
+        const searchValidationError = validatePrisonerSearch({
+          searchTerm: postValues.incidentDetails.currentAssociatedPrisonerNumber,
+          inputId: postValues.incidentDetails.currentIncidentRoleSelection,
+        })
+        if (searchValidationError) {
+          // Could stash to session and redirect here
+          const pageData = await this.getPageDataOnPost(postValues, user)
+          return renderData(res, pageData, searchValidationError)
+        }
+        return redirectToSearchForPersonPage(res, postValues.incidentDetails.currentAssociatedPrisonerNumber)
+      }
+    }
+
+    const validationError = validateForm({
+      incidentDate: postValues.incidentDetails?.incidentDate,
+      locationId: postValues.incidentDetails?.locationId,
+      incidentRole: postValues.incidentDetails?.currentIncidentRoleSelection,
+      associatedPrisonersNumber: postValues.incidentDetails?.currentAssociatedPrisonerNumber,
+    })
+    if (validationError) {
+      // Could stash to session and redirect here
+      const pageData = await this.getPageDataOnPost(postValues, user)
+      return renderData(res, pageData, validationError)
+    }
+
+    const incidentDetailsToSave = postValues.incidentDetails
+
+    try {
```

Out of interest - why would we get an exception here? Should we just let it throw and get the sorry screen?

---

I was preserving the existing functionality - but it does things like set up a specific URL for the Sorry page on error